packages/webview_flutter/example/lib/main.dart

```diff
@@ -47,3 +57,52 @@ class SampleMenu extends StatelessWidget {
     );
   }
 }
+
+class NavigationControls extends StatelessWidget {
+  const NavigationControls(this._webViewControllerFuture)
+      : assert(_webViewControllerFuture != null);
+
+  final Future<WebViewController> _webViewControllerFuture;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: <Widget>[
+        FutureBuilder<WebViewController>(
+          builder: (BuildContext context,
+                  AsyncSnapshot<WebViewController> snapshot) =>
+              IconButton(
+                  icon: const Icon(Icons.arrow_back_ios),
+                  onPressed: () async {
+                    final WebViewController controller = snapshot.data;
+                    final bool canGoBack = await controller.canGoBack();
+                    if (canGoBack) {
+                      controller.goBack();
+                    } else {
+                      Scaffold.of(context).showSnackBar(const SnackBar(
+                          content: Text("No back history item")));
+                    }
+                  }),
+          future: _webViewControllerFuture,
+        ),
+        FutureBuilder<WebViewController>(
```

Comment:
nit: I think it will be simpler to use a single FutureBuilder that wraps the entire row.